spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/JdbcAggregateJobQueryDao.java

```diff
@@ -830,4 +836,121 @@ private boolean determineSupportsRowNumberFunction(DataSource dataSource) {
 		}
 		return false;
 	}
+
+	/**
+	 * A {@link SqlPagingQueryProviderFactoryBean} specialization that overrides the {@code Oracle} paging
+	 * {@link SafeOraclePagingQueryProvider provider} with an implementation that properly handles sort aliases.
+	 * <p><b>NOTE:</b> nested within the aggregate DAO as this is the only place that needs this specialization.
+	 */
+	static class SafeSqlPagingQueryProviderFactoryBean extends SqlPagingQueryProviderFactoryBean {
+
+		private DataSource dataSource;
+
+		@Override
+		public void setDataSource(DataSource dataSource) {
+			super.setDataSource(dataSource);
+			this.dataSource = dataSource;
+		}
+
+		@Override
+		public PagingQueryProvider getObject() throws Exception {
+			PagingQueryProvider provider = super.getObject();
+			if (provider instanceof OraclePagingQueryProvider) {
+				provider = new SafeOraclePagingQueryProvider((AbstractSqlPagingQueryProvider) provider, this.dataSource);
+			}
+			return provider;
+		}
+
+	}
+
+	/**
+	 * A {@link AbstractSqlPagingQueryProvider paging provider} for {@code Oracle} that works around the fact that the
+	 * Oracle provider in Spring Batch 4.x does not properly handle sort aliases when using nested {@code ROW_NUMBER}
+	 * clauses.
+	 */
+	static class SafeOraclePagingQueryProvider extends AbstractSqlPagingQueryProvider {
```

Comment:
NOTE: The whole reason this exists is to get around the issues that @klopfdreh found in https://github.com/spring-cloud/spring-cloud-dataflow/issues/5524#issuecomment-1778630171. 

@cppwfs should we raise these as issues in Batch 4.x ? I am surprised they have not been reported already. Is 4.x EOL or would it make sense to update these in Batch 4.x? If we do, then we should contribute this fix and then remove this patched code here once the proper fix is available.

Reply:
Based on the additional failures in DB2 and SQLServer I am not convinced this is an issue in Batch as much as an issue of our usage.